### PR TITLE
[CALCITE-3442] In ElasticSearch adapter, set `stored_fields = _none_` to prohibit FetchPhase get involved 

### PR DESCRIPTION
**Pull request by wuyunfeng**
Set `stored_fields = _none` to prohibit `FetchPhase` get be involved and improve performance at some agg scenario for elasticsearch-adapter

### CODE REVIEW

**Comment by jinxing64** [comment]
hi @wuyunfeng 
You can run `mvn clean install` in local to test your PR quickly before pushing

**Comment by wuyunfeng** [comment]
@jinxing64 OK

**Comment by wuyunfeng** [comment]
@jinxing64

File: elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/AggregationTest.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ public void countStar() {
         .query("select count(*) from view")
         .queryContains(
             ElasticsearchChecker.elasticsearchChecker(
-                "_source:false, size:0, track_total_hits:true"))
+                "_source:false, 'stored_fields': '_none_', size:0,  track_total_hits:true"))
         .returns("EXPR$0=3\n");
```
Comment (danny0405) [inline elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/AggregationTest.java]: Unnecessary whitespace after `size:0,`

File: elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java [hunk 2]
```diff
@@ -186,6 +186,11 @@ String scriptedFieldPrefix() {
     query.put("_source", false);
     query.put("size", 0);
     query.remove("script_fields");
+    // set _source = false and size = 0, `FetchPhase` would still be executed
+    // to fetch the metadata fields and visit the Lucene stored_fields,
+    // which would lead to performance declined dramatically.
+    // `stored_fields = _none` can prohibit such behavior entirely
+    query.put("stored_fields", "_none_");
```
Comment (danny0405) [inline elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java]: Sounds reasonable, do you think we can add a test case for this ? Some code to  illustrates that the `FetchPhase` you mentioned has indeed been triggered without your fix.

Comment (wuyunfeng) [reply elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java]: hi, @danny0405  . Thanks for review.  I have worked on Doris On Elasticsearch (https://github.com/apache/incubator-doris/pull/2047).  I found this problem few months ago, If the query sent to Elasticsearch without `stored_fields : _none_` the fetch phase must be executed in order to get some meta data fields 

I submit a PR for es can implicitly promote this behavior: https://github.com/elastic/elasticsearch/pull/41155 

but they told me add `stored_fields : _none_`  to the query  can reach this goal.

Elasticsearch X-Pack SQL, they also add `stored_fields : _none_` to the query for Aggregation Transform

File: elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/AggregationTest.java (hunk 1 above)
Comment (wuyunfeng) [reply elasticsearch/src/test/java/org/apache/calcite/adapter/elasticsearch/AggregationTest.java]: done

**Comment by wuyunfeng** [comment]
@danny0405 

X-Pack SQL translate API can prove this:

```
POST /_sql/translate
{
    "query": "SELECT k1,count(*) FROM test GROUP BY k1,k2",
    "fetch_size": 100
}
```
the result:


```
{
   "size": 0,
   "_source": false,
   "stored_fields": "_none_",
   "aggregations": {
      "groupby": {
         "composite": {
            "size": 100,
            "sources": [
               {
                  "612": {
                     "terms": {
                        "field": "k1",
                        "missing_bucket": true,
                        "order": "asc"
                     }
                  }
               },
               {
                  "613": {
                     "terms": {
                        "field": "k2",
                        "missing_bucket": true,
                        "order": "asc"
                     }
                  }
               }
            ]
         }
      }
   }
}

```

File: elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java (hunk 2 above)
Comment (wuyunfeng) [reply elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java]: All logic can see at `FetchPhase`:

https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

`FetchPhase#execute`

Comment (danny0405) [reply elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchTable.java]: Cool, thanks for the explanation ~

**Comment by wuyunfeng** [comment]
@danny0405 Thanks for your review.

**Comment by asereda-gs** [comment]
@wuyunfeng thanks for this PR.

Changes also look good to me. Can you please squash commits into one and add your full name into commit message ?
